src/StreamrClient.js

```diff
@@ -46,6 +47,7 @@ export default class StreamrClient extends EventEmitter {
         }
 
         this.session = new Session(this, options.auth)
+        this.signer = (options.auth && options.auth.publishWithSignature) ? new Signer(options.auth) : undefined
```

Comment:
The option feels like it's in the wrong place under `auth`. Would be cool to also have care-free default functionality. How about something like this:

```
new StreamrClient({
    auth: {
        // as before
    },
    publishWithSignature: 'auto', // valid values: 'auto' (default), 'always', 'never'
})
```

- 'auto': sign messages if possible (key or provider given)
- 'always': always sign messages (throw if not possible)
- 'never': never sign messages (even if possible)